tests/data_files/Makefile

```diff
@@ -1602,3 +1602,5 @@ clean:
 neat: clean
 	rm -f $(all_final)
 .PHONY: clean neat
+
+.INTERMEDIATE: $(all_intermediate)
```

Comment:
We should probably list those files in `.gitignore`. Or maybe we should check them all in, I'm not sure. The only difference between “intermediate” and “final” files is that the intermediate files are not used in any test case, but does that really matter?

I don't want them listed as `.INTERMEDIATE` because that causes `make` to delete them as soon as they're created, which is a problem when you're working on the final files. Suppose you have a dependency chain A→B→C where B is intermediate in the sense of make. You run `make C`, that creates B and C but deletes B. You then run `make C` again, and that re-creates B and C since B doesn't exist. This is especially problematic when the generation of B is not reproducible (e.g. depends on the date), which is the case for some of the files there.

Reply:
> You then run make C again, and that re-creates B and C since B doesn't exist. This is especially problematic when the generation of B is not reproducible (e.g. depends on the date), which is the case for some of the files there.

When run `make C`, B and C won't be create again. For debug purpose, we can run `make B`, B won't be deleted, and if B exists before `make C`, it won't be removed also. 

Below is my test result
```
haoyu01@haoyu01-01-x86-vm:~/workspace_ceph/mbedtls$ make -f t.mk C
touch B
touch C
rm B
haoyu01@haoyu01-01-x86-vm:~/workspace_ceph/mbedtls$ make -f t.mk B
touch B
haoyu01@haoyu01-01-x86-vm:~/workspace_ceph/mbedtls$ make -f t.mk C
touch C
haoyu01@haoyu01-01-x86-vm:~/workspace_ceph/mbedtls$ 
```
And t.mk is
```makefile
:B
A:
B:A

%:
	touch $@
.INTERMEDIATE: B
```

Reply:
So I think we should list those files in `.gitignore` and `.INTERMEDIATE` . That's not a big problem, just keep a clean folder